test/typescript/almin-test.ts

```diff
@@ -0,0 +1,110 @@
+import {
+    Context,
+    Store,
+    StoreGroup,
+    Dispatcher,
+    UseCase,
+    Payload,
+    WillExecutedPayload,
+    DidExecutedPayload,
+    CompletedPayload,
+    ErrorPayload,
+    DispatcherPayloadMeta
+} from "../../src/index";
+// Dispatcher
+const dispatcher = new Dispatcher();
+// Store
+interface AState {
+    a: number
+}
+class AStore extends Store {
+    state: AState;
+
+    constructor() {
+        super();
+        this.state = {
+            a: 1
+        }
+    }
+
+    getState() {
+        return {
+            A: this.state
+        };
+    }
+}
+interface BState {
+    b: number;
+}
+class BStore extends Store {
+    state: BState;
+
+    constructor() {
+        super();
+        this.state = {
+            b: 2
+        }
+    }
+
+    getState() {
+        return {
+            B: this.state
+        };
+    }
+}
+// StoreGroup
+interface StoreState {
+    A: AState;
+    B: BState;
+}
+const storeGroup = new StoreGroup([
+    new AStore(),
+    new BStore()
+]);
+// Context
+const context = new Context({
+    dispatcher,
+    store: storeGroup
+});
+// Context - life cycle
+context.onWillExecuteEachUseCase((payload: WillExecutedPayload, meta: DispatcherPayloadMeta) => {
+    console.log(payload.args, meta);
+});
+context.onDidExecuteEachUseCase((payload: DidExecutedPayload, meta: DispatcherPayloadMeta) => {
+    console.log(payload.value, meta);
+});
+context.onCompleteEachUseCase((payload: CompletedPayload, meta: DispatcherPayloadMeta) => {
+    console.log(payload.value, meta);
+});
+context.onErrorDispatch((payload: ErrorPayload, meta: DispatcherPayloadMeta) => {
+    console.log(payload.error, meta);
+});
+context.onDispatch((payload: Payload, meta: DispatcherPayloadMeta) => {
+    console.log(payload.type, meta);
+});
+
+// UseCase
+class ChildUseCase extends UseCase {
+    execute(value: string) {
+        this.dispatch({
+            type: "ChildUseCase",
+            value
+        });
+    }
+}
+class ParentUseCase extends UseCase {
+    execute(value: string) {
+        // TODO: improve `execute` signature
+        return this.context.useCase(new ChildUseCase()).execute(value);
+    }
+}
+const parentUseCase = new ParentUseCase();
+// UseCase - execute
+context.useCase(parentUseCase).execute("value").then(() => {
+    // TODO: Can improve without StoreState?
+    const state = context.getState<StoreState>();
```

Comment:
> Could we get it more easily?

Umm, this is the one of weakpoints of the current TypeScript.  Fundamentally, we need to wait [Variadic Type Proposal for TypeScript](https://github.com/Microsoft/TypeScript/issues/5453) to resolve this problem with straight way.

As other approach, we can use multiple overloading. This technique is used in rxjs' `zip`, `combineLatest`, and `Object.assign()`.

```typescript
    getState<T1, T2>(): T1 & T2;
    getState<T1, T2, T3>(): T1 & T2 & T3;
    getState<T1, T2, T3, T4>(): T1 & T2 & T3 & T4;
    getState<T1, T2, T3, T4, T5>(): T1 & T2 & T3 & T4 & T5;
    getState<T1, T2, T3, T4, T5, T6>(): T1 & T2 & T3 & T4 & T5 & T6;
    getState<T>(): T {
        // actual implementation.
    }
```

And we can add type param `T` to `StoreLike` like `StoreLike<T>`. Then we can make `StoreLike.getState()` to `getState(): T`. I feel this approach is more straight way than overloading. But we also need to consider if we bind some `StoreLike` into an other container object.